### PR TITLE
upgrade to zig 0.12.0

### PR DESCRIPTION
**Pull request by softprops**
follow up from [discord  message](https://discord.com/channels/605571803288698900/1247575320823988227/1247643658149171361)

### CODE REVIEW

File: example.zig [hunk 1]
```diff
@@ -1,5 +1,5 @@
 const std = @import("std");
-const benchmark = @import("benchmark.zig");
+const benchmark = @import("benchmark");
```
Comment (silversquirl) [inline example.zig:2]: Could you update the code in the readme the same way?

Comment (softprops) [reply example.zig:2]: done

File: README.md [hunk 2]
```diff
@@ -4,13 +4,13 @@ This is a tiny microbenchmark library for Zig, designed to be easy to use and ve
 
 ## Usage
 
-You can either manually copy `benchmark.zig` into your project, or add it as a dependency using the Zig package manager.
+You can either manually copy `benchmark` into your project, or add it as a dependency using the Zig package manager.
```
Comment (silversquirl) [inline README.md]: the `.zig` should stay in this line, though maybe clarifying that it's the `benchmark.zig` *file* that you can copy directly into your project would be good :)

Comment (softprops) [reply README.md]: updated wording

**Comment by silversquirl** [comment]
Thanks!

**Pull request merged**